operator/node.go

```diff
@@ -128,64 +111,93 @@ func (n *operatorNode) StartEth1(syncOffset *eth1.SyncOffset) error {
 	return nil
 }
 
-func (n *operatorNode) onDuty(duty *ethpb.DutiesResponse_Duty) {
-	slots := collectSlots(duty)
-	if len(slots) == 0 {
-		n.logger.Debug("no slots found for the given duty")
+func (n *operatorNode) onDuty(duty *beacon.Duty) {
+	if uint64(duty.Slot) < n.getEpochFirstSlot(n.genesisEpoch) {
+		// wait until genesis epoch starts
+		n.logger.Debug("skipping slot, lower than genesis",
+			zap.Uint64("genesis_slot", n.getEpochFirstSlot(n.genesisEpoch)),
+			zap.Uint64("slot", uint64(duty.Slot)))
 		return
 	}
 
-	for _, slot := range slots {
-		if slot < n.getEpochFirstSlot(n.genesisEpoch) {
-			// wait until genesis epoch starts
-			n.logger.Debug("skipping slot, lower than genesis",
-				zap.Uint64("genesis_slot", n.getEpochFirstSlot(n.genesisEpoch)),
-				zap.Uint64("slot", slot))
+	currentSlot := uint64(n.getCurrentSlot())
+	logger := n.logger.
```

Comment:
@nivBlox not sure this log needs to be here, why not log it on the validator level?

Reply:
in order to provide data for when inserting duty to queue. validator level is when the duty is executing